filter/command.go

```diff
@@ -58,16 +58,24 @@ func (o Options) Run() error {
 		matches = matchAll(choices)
 	}
 
+	if o.NoLimit {
+		o.Limit = len(choices)
+	}
+
 	p := tea.NewProgram(model{
-		choices:        choices,
-		indicator:      o.Indicator,
-		matches:        matches,
-		textinput:      i,
-		viewport:       &v,
-		indicatorStyle: o.IndicatorStyle.ToLipgloss(),
-		matchStyle:     o.MatchStyle.ToLipgloss(),
-		textStyle:      o.TextStyle.ToLipgloss(),
-		height:         o.Height,
+		choices:            choices,
+		indicator:          o.Indicator,
+		matches:            matches,
+		textinput:          i,
+		viewport:           &v,
+		indicatorStyle:     o.IndicatorStyle.ToLipgloss(),
+		selectionMarkStyle: o.SelectionMarkStyle.ToLipgloss(),
+		selectionMark:      o.SelectionMark,
+		matchStyle:         o.MatchStyle.ToLipgloss(),
+		textStyle:          o.TextStyle.ToLipgloss(),
+		height:             o.Height,
+		multiSelection:     make(map[string]interface{}),
```

Comment:
I think we should change this to `map[string]bool`, what do you think?

```suggestion
		multiSelection:     make(map[string]bool),
```

Reply:
I don't see the benefits on using a boolean since it's basically meaningless which values the map holds. Anyways I don't have any strong opinion about it and I might miss the purpose behind this, thus I can make a new commit with that change reflected on the code.

Reply:
Hmm, I see your point. Do you know if there are memory benefits of having this `map` map to pointless values rather than booleans? I would guess that there may be some which might be slightly advantageous with large filters.

Reply:
To my surprise it actually is counter-productive to use the interface{}. On second thought, interface{} can effectively hold any value, and therefore here is the explanation on its slowness.

I'll leave here the result of my simple benchmark.
```
goos: linux
goarch: amd64
pkg: bench
cpu: AMD Ryzen 7 PRO 4750U with Radeon Graphics
BenchmarkInterface-16            4549672               251.9 ns/op
BenchmarkEmptyStruct-16          6298020               172.5 ns/op
BenchmarkBool-16                 5262289               190.8 ns/op
PASS
ok      bench   4.955s
```
Test file:
```go
package main

import "testing"

func BenchmarkInterface(b *testing.B) {
	m := map[int]interface{}{}
	for i := 0; i < b.N; i++ {
		m[i] = nil
	}
}

func BenchmarkEmptyStruct(b *testing.B) {
	m := map[int]struct{}{}
	for i := 0; i < b.N; i++ {
		m[i] = struct{}{}
	}
}

func BenchmarkBool(b *testing.B) {
	m := map[int]bool{}
	for i := 0; i < b.N; i++ {
		m[i] = true
	}
}
```

Reply:
Sweet! Looks like we should be using an empty `struct{}` then? I think that makes sense to me. Thank you for running these benchmarks!

Reply:
Worth also just double checking if theres a difference between setting the boolean to `false` rather than `true` in case that makes any difference.

Reply:
Good point on the false vs true possible difference.
I just tested it and it did not seem to make any difference:
```
goos: linux
goarch: amd64
pkg: bench
cpu: AMD Ryzen 7 PRO 4750U with Radeon Graphics
BenchmarkInterface-16            3932340               293.8 ns/op
BenchmarkEmptyStruct-16          6849700               166.6 ns/op
BenchmarkBool-16                 6222806               176.5 ns/op
PASS
ok      bench   6.053s
```

Reply:
Awesome! Thank you so much, I say we should use an empty `struct{}` in that case

Reply:
Perfect, I'll work on a commit that fixes all the changes proposed in the review then.

Reply:
❤️